### PR TITLE
feature/place_checkin_api_end_point

### PR DESCRIPTION
**Pull request by fadykstas**
setCheckin: action, request, response, route, httpRequest, criteria, api, tests

### CODE REVIEW

File: app/Http/Controllers/Api/Places/PlaceCheckinController.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+class PlaceCheckinController extends ApiController
+{
+    private $setCheckinAction;
+    private $userId;
+
+    public function __construct(SetCheckinAction $setCheckinAction )
+    {
+        $this->setCheckinAction = $setCheckinAction;
+        if (Auth::check()) {
```
Comment (nemoipaha) [inline app/Http/Controllers/Api/Places/PlaceCheckinController.php]: Oh noo) Move this inside action class.

File: app/Http/Controllers/Api/Places/PlaceCheckinController.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    public function setCheckin(SetCheckinHttpRequest $httpRequest) : JsonResponse
+    {
+        try {
+            $response = $this->setCheckinAction->execute(
+                new SetCheckinRequest(
+                    $this->userId,
+                    $httpRequest->place_id
+                )
+            );
+        } catch (\Exception $ex) {
```
Comment (nemoipaha) [inline app/Http/Controllers/Api/Places/PlaceCheckinController.php]: Catch your custom exception in action like PlaceNotFoundException

File: app/Http/Requests/Place/Checkin/SetCheckinHttpRequest.php [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+{
+    public function authorize() : bool
+    {
+        return true;
+    }
+
+    public function rules() : array
+    {
+        return [
+            'place_id' => 'required|integer|min:1',
```
Comment (nemoipaha) [inline app/Http/Requests/Place/Checkin/SetCheckinHttpRequest.php]: required is enough.

File: app/Actions/Place/Checkin/SetCheckinAction.php [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+        $checkin = $this->repository->findByCriteria(
+            new UserIdAndPlaceIdCriteria(
+                $userId,
+                $placeId
+            )
+        )->first();
+
+
+        $checkin = $checkin ? : new Checkin([
```
Comment (nemoipaha) [inline app/Actions/Place/Checkin/SetCheckinAction.php]: User can checkin one place few times. This validation can be removed. Just save new checkin model

File: app/Actions/Place/Checkin/SetCheckinAction.php [hunk 5]
```diff
@@ -4,26 +4,34 @@
...
 {
     protected $repository;
     protected $placeRepository;
+    private $userId;
 
     public function __construct(CheckinRepositoryInterface $repository, PlaceRepositoryInterface $placeRepository)
     {
         $this->repository = $repository;
         $this->placeRepository = $placeRepository;
+        if (Auth::check()) {
```
Comment (nemoipaha) [inline app/Actions/Place/Checkin/SetCheckinAction.php]: No need for Auth::check() , this SetCheckinAction is protected and only authenticated user can request it.

**Comment by fadykstas** [comment]
related #123

File: routes/api.php [hunk 6]
```diff
@@ -99,8 +99,22 @@
...
             ->name('place.features.deleteFeature');
 
-
         Route::post('/users/me/checkins', 'Api\Places\PlaceCheckinController@setCheckin')
             ->name('user.me.checkin');
+      
+        Route::post('/places/rating', 'Api\Places\PlaceRatingController@setRating')
+            ->name('place.rating.setPlaceRating');
+
+        Route::get('/places/rating/place/{id}', 'Api\Places\PlaceRatingController@getPlaceRatingAvg')
```
Comment (nemoipaha) [inline routes/api.php]: Better is `get /places/{id}/rating/`

**Pull request merged**